src/rules/order.js

```diff
@@ -198,58 +204,113 @@ function makeImportDescription(node) {
   return 'import';
 }
 
-function fixOutOfOrder(context, firstNode, secondNode, order) {
+function fixOutOfOrder(context, firstNode, secondNode, order, named) {
   const sourceCode = context.getSourceCode();
 
-  const firstRoot = findRootNode(firstNode.node);
+  const {
+    firstRoot,
+    secondRoot,
+  } = named ? {
+    firstRoot: firstNode.node,
+    secondRoot: secondNode.node,
+  } : {
+    firstRoot: findRootNode(firstNode.node),
+    secondRoot: findRootNode(secondNode.node),
+  };
+
   const firstRootStart = findStartOfLineWithComments(sourceCode, firstRoot);
   const firstRootEnd = findEndOfLineWithComments(sourceCode, firstRoot);
 
-  const secondRoot = findRootNode(secondNode.node);
   const secondRootStart = findStartOfLineWithComments(sourceCode, secondRoot);
   const secondRootEnd = findEndOfLineWithComments(sourceCode, secondRoot);
-  const canFix = canReorderItems(firstRoot, secondRoot);
 
-  let newCode = sourceCode.text.substring(secondRootStart, secondRootEnd);
-  if (newCode[newCode.length - 1] !== '\n') {
-    newCode = `${newCode}\n`;
+  if (firstNode.displayName === secondNode.displayName) {
+    for (const node of [firstNode, secondNode]) {
+      if (node.alias) {
+        node.displayName = `${node.displayName} as ${node.alias}`;
+      }
+    }
```

Comment:
i'm a bit confused here why we're just overwriting the first node's alias when the second node has one - either way, let's avoid for-of here, it's unnecessary for 2 hardcoded items and it's slower than other forms of iteration.

Reply:
Alright, will do!